daffodil-runtime1/src/main/scala/org/apache/daffodil/layers/AISTransformer.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.layers
+
+import org.apache.daffodil.schema.annotation.props.gen.LayerLengthKind
+import org.apache.daffodil.schema.annotation.props.gen.LayerLengthUnits
+import org.apache.daffodil.util.Maybe
+import org.apache.daffodil.processors.TermRuntimeData
+import org.apache.daffodil.processors.LayerLengthInBytesEv
+import org.apache.daffodil.processors.LayerBoundaryMarkEv
+import org.apache.daffodil.processors.LayerCharsetEv
+import org.apache.daffodil.processors.parsers.PState
+import org.apache.daffodil.processors.unparsers.UState
+import org.apache.daffodil.processors.charset.AIS_PAYLOAD_ARMORING
+import java.nio._
+import java.nio.charset._
+import java.io._
+import org.apache.commons.io.IOUtils
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.io.BoundaryMarkLimitingStream
+import org.apache.daffodil.io.LayerBoundaryMarkInsertingJavaOutputStream
+
+object AISTransformer {
```

Comment:
Make naming more specific: AIS to AIS_ASCII_ARMORING or similar. The name of the layer transform should probably also be not "ais" but "aisASCIIArmoring"